plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt

```diff
@@ -450,6 +485,18 @@ class DefaultPsiToDocumentableTranslator(
             )
         }
 
+        /**
+         * `isPhysical` detects the virtual declarations without real sources.
+         * Otherwise, [PsiDocumentableSource] initialization will fail: non-physical declarations doesn't have `virtualFile`.
+         * The check is required since in [parseConstructors] we create a virtual method for default constructor.
```

Comment:
nitpick:

The last line is quite specific and it can rot with time if the implementation of `parseConstructors` changes and we forget to update this comment (we probably will forget it)

I'd change it to something more generic, like `This check protects from accidentally requesting sources for synthetic / virtual declarations.`

And it can be helpful to move the comment closer to `isPhysical` and make it an implementation comment (`// text`), since more generic logic for parsing sources can be added to this function in the future.